frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -39,9 +42,46 @@ export default function UserAnswerPage() {
   const lastPage =
     questions && Array.isArray(questions) ? questions.length + 1 : 0;
 
+  const totalQuestion =
+    questions && Array.isArray(questions) ? questions.length : 0;
+  const passingScore = 75;
+
+  useEffect(() => {
+    setFailed(scorePercentage < passingScore ? true : false);
+    setPassed(scorePercentage >= passingScore ? true : false);
+  }, [scorePercentage]);
+
+  useEffect(() => {
+    const data = {
+      score,
+      totalQuestion,
+    };
+    const calculate = async (data) => {
+      try {
+        const result = await calculateScore(data);
```

Comment:
dont create a variable if you are only using it once
```suggestion
    const calculate = async (data) => {
      try {
        const result = await calculateScore({
      score,
      totalQuestion,
    });
```